packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -246,7 +247,7 @@ export function createInsightsMiddleware<
         let insightsClientWithLocalCredentials =
           insightsClient as InsightsClientWithLocalCredentials;
 
-        if (isModernInsightsClient(insightsClient)) {
+        if (supportsOptionalInitMethod(insightsClient)) {
```

Comment:
I liked modern more, as it is optional init + credentials through arguments. It's fine like this too of course

Reply:
Either let's convey this with an explicit name, or use a comment, wdyt?

Reply:
I guess technically "supportsOptionalInitMethod" covers both usages of the function, but I wonder if it should be "supportsCredentialsInPayload" or similar?

Reply:
The function is called in both places, the first one being to check if the optional `init` method is supported and the other one is to check if the credentials are supported as extra params.

So maybe we should keep a generic function name like `isModernInsightsClient()` that encapsulates both behavior and add comments at both places of use, WDYT?

Reply:
I think `modern` makes sense, but a jsdoc comment to the function saying it has: "partial init" and "credentials via event" maybe?

Reply:
Done, thanks!